Makefile

```diff
@@ -98,13 +98,17 @@ brew::
 
 .PHONY: lint_%
 lint:: golangci-lint.ensure lint_deps
+	stat go.work || go work init ./pkg ./sdk/ ./tests
```

Comment:
@RobbieMcKinstry typically the go.work might exist outside this directory.
Like you might have,

```
ws/go.work
ws/pulumi/
ws/pulumi-command/
// ...
```

Where ws/ is not a git repository. 

I think the only reliable way to detect if you're in a go.work space is `go env GOWORK`.
That's empty if you're not in a workspace and reports the path to the go.work file otherwise.

Per https://go.dev/blog/get-familiar-with-workspaces,

> Workspace mode is enabled if the GOWORK variable names a path to a file that ends in .work. To determine which go.work file is being used, run go env GOWORK. The output is empty if the go command is not in workspace mode.

Optionally, if you'd like to keep things simple—this go.work business is needed only for the lint job. 
We can create the go.work file in ci-lint.yml for only the "Lint Go code" step, and nothing else.

Reply:
Follow up on this. I'm suggesting something like:

```yaml
      - name: Lint
        run: |
          go work init ./pkg ./sdk ./tests
          make lint
```

```make
.PHONY: lint_%
lint:: golangci-lint.ensure

# If we're in a Go workspace, assume that pkg, sdk, and tests are all covered.
ifeq ($(shell go env GOWORK),)
lint:: lint_pkg lint_sdk lint_tests
else
lint::
	golangci-lint run ./pkg/... ./sdk/... ./tests/... --timeout 7m
endif

lint_pkg: lint_deps
	cd pkg && golangci-lint run -c ../.golangci.yml --timeout 5m
lint_sdk: lint_deps
	cd sdk && golangci-lint run -c ../.golangci.yml --timeout 5m
lint_tests: lint_deps
	cd tests && golangci-lint run -c ../.golangci.yml --timeout 5m
```

It's not perfect, but it'll do what we want.

Reply:
Nit - do we ant `./pkg/...` or `./pkg` there?

Reply:
I believe you're passing a path to golangci-lint, it wants a Go import pattern, not a directory. `./pkg` will only lint the Go files inside the pkg package.